addons/account/models/account_move.py

```diff
@@ -3715,14 +3715,16 @@ def _check_constrains_account_id_journal_id(self):
 
     @api.constrains('account_id', 'tax_ids', 'tax_line_id', 'reconciled')
     def _check_off_balance(self):
-        for line in self:
-            if line.account_id.internal_group == 'off_balance':
-                if any(a.internal_group != line.account_id.internal_group for a in line.move_id.line_ids.account_id):
-                    raise UserError(_('If you want to use "Off-Balance Sheet" accounts, all the accounts of the journal entry must be of this type'))
-                if line.tax_ids or line.tax_line_id:
-                    raise UserError(_('You cannot use taxes on lines with an Off-Balance account'))
-                if line.reconciled:
-                    raise UserError(_('Lines from "Off-Balance Sheet" accounts cannot be reconciled'))
+        off_balance_line = self.search([('id', 'in', self.ids), ('account_id.internal_group', '=', 'off_balance'), ])
+        non_off_balance_line = self.search([('id', 'in', self.ids), ('account_id.internal_group', '!=', 'off_balance'), ])
+        balance_in_off_balance = sum(l.debit - l.credit for l in off_balance_line)
+        balance_in_non_off_balance = sum(l.debit - l.credit for l in non_off_balance_line)
+        if not self[0].company_currency_id.is_zero(balance_in_off_balance) or not self[0].company_currency_id.is_zero(balance_in_non_off_balance):
+            raise UserError(_('If you want to use "Off-Balance Sheet" accounts, the "Off-Balance Sheet" items must be balanced.'))
+        if off_balance_line and any(l.tax_ids or l.tax_line_id for l in off_balance_line):
```

Comment:
It would be better to use off_balance_lines (as there are several).

This can be:
```suggestion
        if off_balance_lines and off_balance_lines.filtered(lambda l: l.tax_ids or l.tax_line_id):
```

Reply:
This is no longer valid due to recent changes.